### PR TITLE
Fix sidebar look after import_export

### PR DESCRIPTION
**Pull request by unho**
This fixes issue #3640

- Added a new #sidebar-content div to enclose sidebar content in order
  to correctly hide the whole sidebar except the toggle handle
- Moved padding and margin from specific content divs (#staticpage and
  #import-export-sidebar) to that new div
- Changed some texts to provide more useful hints to the users
- Use placeholder instead of input value
- Removed grey background for import-export
- Added some JavaScript to avoid user typing on the fake upload input

### CODE REVIEW

**Comment by unho** [comment]
@translate/house @translate/evernote Please review. Tests are also welcome. Original issue is https://github.com/translate/pootle/issues/3640

**Comment by jleclanche** [comment]
lgtm, didnt test

File: pootle/apps/pootle_translationproject/views.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 from staticpages.models import StaticPage
 
 
-ANN_COOKIE_NAME = 'project-announcements'
+ANN_COOKIE_NAME = 'sidebar-cookie'
```
Comment (julen) [inline pootle/apps/pootle_translationproject/views.py]: If renaming, mind doing it to something like `pootle-overview-sidebar`? Feel free to push that to master directly.

Comment (julen) [inline pootle/apps/pootle_translationproject/views.py]: And at the same time `ANN_COOKIE_NAME` should be renamed as well; `SIDEBAR_COOKIE_NAME` or similar.

**Comment by julen** [comment]
FTR, the layout was fixed in 8f407a91, this PR makes some minor visual adjustments. I have added some related comments in #3640.

File: pootle/apps/pootle_translationproject/views.py (hunk 1 above)
Comment (unho) [inline pootle/apps/pootle_translationproject/views.py]: @julen Ok. I think this can safely go in a separate commit.

**Comment by dwaynebailey** [comment]
@unho from what I understand this doesn't address the concerns in #3640 about the upload widget.

**Comment by unho** [comment]
@iafan @julen @dwaynebailey Updated PR to address your concerns. Please review and test again.

**Comment by dwaynebailey** [comment]
@unho I tested specifically the changed upload behaviour and it's gtm

File: pootle/static/css/style.css [hunk 2]
```diff
@@ -2121,15 +2127,19 @@ span.email .i
...
+{
+    margin: 0;
+    padding: 0;
 }
 
-#upload-form {
-    margin-top: .5em;
+.action-link
+{
+    display: block;
```
Comment (julen) [inline pootle/static/css/style.css]: why not make it more intuitive and wrap the inline element within a block element such as `<ul><li>`, `<p>`, or `<div>`?

**Comment by julen** [comment]
* The link underline goes beyond the text boundaries, so the initial whitespace also displays an underline which is not nice. You can check how we work around that in the overview file/dir listing icons.
* When there are other hooked up elements in the sidebar, it'd be good to display some separator to make it clear the contents are unrelated.
screenshot

 
These two small issues aside looks good to merge

**Comment by unho** [comment]
@julen Updated to address your latest concerns. I would appreciate if you can quickly review and test and maybe gtm the PR.

File: pootle/templates/browser/overview.html [hunk 3]
```diff
@@ -78,13 +78,21 @@ <h4 class="clear">{{ item.title }}</h4>
...
+  <div id="sidebar-content">
+    {% block sidebar_content %}
 
-  {% if announcement %}
-  {% include "staticpages/_body.html" with page=announcement %}
-  {% endif %}
+    {% if announcement %}
+    {% comment %}Dirty hack to place a separator while we don't have a way to
+    tell if after an existing item we have another one.{% endcomment %}
+    {% if display_download or upload_form %}
```
Comment (julen) [inline pootle/templates/browser/overview.html]: I'd rather not couple the sidebar to 3rd party specifics and live without a separator instead.

Comment (julen) [inline pootle/templates/browser/overview.html]: Having said that, it's still possible to add the separator by having the content for each content type within a block, and using CSS have all those blocks, except the last one, to render a `border-bottom`.
